pkg/reconciler/approvaltask/utils.go

```diff
@@ -70,47 +75,35 @@ func initializeCustomRun(ctx context.Context, run *v1beta1.CustomRun) {
 	}
 }
 
-func (r *Reconciler) getOrCreateApprovalTask(ctx context.Context, run *v1beta1.CustomRun) (*metav1.ObjectMeta, *v1alpha1.ApprovalTaskSpec, error) {
-	logger := logging.FromContext(ctx)
-	approvalTaskMeta := metav1.ObjectMeta{}
-	approvalTaskSpec := v1alpha1.ApprovalTaskSpec{}
+func getOrCreateApprovalTask(ctx context.Context, approvaltaskClientSet versioned.Interface, run *v1beta1.CustomRun) (*v1alpha1.ApprovalTask, error) {
+	approvalTask := v1alpha1.ApprovalTask{}
 
 	if run.Spec.CustomRef != nil {
 		// Use the k8 client to get the ApprovalTask rather than the lister.  This avoids a timing issue where
 		// the ApprovalTask is not yet in the lister cache if it is created at nearly the same time as the Run.
 		// See https://github.com/tektoncd/pipeline/issues/2740 for discussion on this issue.
-		tl, err := r.approvaltaskClientSet.OpenshiftpipelinesV1alpha1().ApprovalTasks(run.Namespace).Get(ctx, run.Name, metav1.GetOptions{})
+		tl, err := approvaltaskClientSet.OpenshiftpipelinesV1alpha1().ApprovalTasks(run.Namespace).Get(ctx, run.Name, metav1.GetOptions{})
 		if err != nil {
 			if errors.IsNotFound(err) {
-				approvalTask := &v1alpha1.ApprovalTask{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: run.Name,
-					},
-					Spec: v1alpha1.ApprovalTaskSpec{
-						Name:     run.Name,
-						Approved: "wait",
-					},
-				}
-
-				tl, err = r.approvaltaskClientSet.OpenshiftpipelinesV1alpha1().ApprovalTasks(run.Namespace).Create(ctx, approvalTask, metav1.CreateOptions{})
+				at, err := createApprovalTask(ctx, approvaltaskClientSet, run)
 				if err != nil {
-					return nil, nil, err
+					return nil, err
 				}
-				logger.Infof("Approval Task %s is created", approvalTask.Name)
+				return &at, nil
 			}
 		}
-		approvalTaskMeta = tl.ObjectMeta
-		approvalTaskSpec = tl.Spec
+		approvalTask = *tl
 	} else if run.Spec.CustomSpec != nil {
 		// FIXME(openshift-pipelines) support embedded spec
-		if err := json.Unmarshal(run.Spec.CustomSpec.Spec.Raw, &approvalTaskSpec); err != nil {
+		if err := json.Unmarshal(run.Spec.CustomSpec.Spec.Raw, &approvalTask.Spec); err != nil {
 			run.Status.MarkCustomRunFailed(v1alpha1.ApprovalTaskRunReasonCouldntGetApprovalTask.String(),
 				"Error retrieving ApprovalTask for Run %s/%s: %s",
 				run.Namespace, run.Name, err)
-			return nil, nil, fmt.Errorf("Error retrieving ApprovalTask for Run %s: %w", fmt.Sprintf("%s/%s", run.Namespace, run.Name), err)
+			return nil, fmt.Errorf("Error retrieving ApprovalTask for Run %s: %w", fmt.Sprintf("%s/%s", run.Namespace, run.Name), err)
 		}
 	}
-	return &approvalTaskMeta, &approvalTaskSpec, nil
+
+	return &approvalTask, nil
```

Comment:
returns empty `ApprovalTask`, when `spec.customRef` and `spec.customSpec` are `nil`.
Is that fine?